arm-software/embedded/CMakeLists.txt

```diff
@@ -887,13 +887,23 @@ add_custom_target(
     COMMAND "${CMAKE_COMMAND}" -E rm -rf unpack
     COMMAND "${CMAKE_COMMAND}" -E make_directory unpack
 )
-add_custom_target(
-    unpack-llvm-toolchain
-    COMMAND "${CMAKE_COMMAND}" -E tar x ${package_filepath}
-    DEPENDS ${package_filepath}
-    USES_TERMINAL
-    WORKING_DIRECTORY unpack
-)
+if(package_filename_extension STREQUAL ".dmg")
+    add_custom_target(
+        unpack-llvm-toolchain
+        COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/scripts/extract_dmg.sh ${package_filepath} ${PACKAGE_FILE_NAME}
```

Comment:
This command line took me some head-scratching to make sense of. From the script below, `$1` is the .dmg file to unpack, and `$2` is a glob pattern? So we're extracting only one thing from the archive and it's called `${PACKAGE_FILE_NAME}`? I guess that must be a top-level directory inside the .dmg containing the rest of the distribution?

Reply:
That's correct. The .dmg file contains an extra symlink to the `Applications` directory, which is there for the drag-and-drop installation experience on Macs, so simply extracting all of the image's contents is probably not the best approach.